### PR TITLE
[HiCache] Batch PP write and load completion sync

### PR DESCRIPTION
**Pull request by luoroger37**
## Motivation

Reduce scheduler overhead for HiCache with pipeline parallelism by batching write and load completion-count synchronization into a single operation.

## Modifications

- Synchronize [write_finish_count, load_finish_count] together for PP > 1.
- Reuse the existing CP/TP reduction and PP propagation path.
- Preserve the existing ACK processing, event synchronization, storage, and lock-release behavior.
- Add unit coverage for PP leader and follower behavior.

## Benchmark

We evaluated the change using a prefill-only shared-prefix workload with 3.5K, 32K, and 128K input lengths and one output token. The benchmark adaptively increased request rate and concurrency to search for the serving throughput limit.

The clearest improvement was observed in the 32K high-pressure workload, where the maximum logical input throughput increased from approximately 494K to 677K tokens/s (+37%).

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :white_check_mark: [Run #32008637682](https://github.com/sgl-project/sglang/actions/runs/32008637682)
Latest PR Test (Extra): :x: [Run #32008637485](https://github.com/sgl-project/sglang/actions/runs/32008637485)

### CODE REVIEW

File: python/sglang/srt/mem_cache/unified_radix_cache.py [hunk 1]
```diff
@@ -1937,8 +1937,18 @@ def check_hicache_events(self) -> None:
         self._drain_async_work()
 
         if self.pp_size != 1:
-            self.writing_check()
-            self.loading_check()
+            finish_counts = torch.zeros(2, dtype=torch.int, device="cpu")
```
Comment (stepinto) [inline python/sglang/srt/mem_cache/unified_radix_cache.py:2035]: I think we could merge this with the else branch (line 1955..1979), which could further more reduce the number of synchronizations.

Comment (luoroger37) [reply python/sglang/srt/mem_cache/unified_radix_cache.py:2035]: IC，I've only considered L2 so far, not L3. It's true that including L3 would further reduce synchronization.

Comment (luoroger37) [reply python/sglang/srt/mem_cache/unified_radix_cache.py:2035]: > I think we could merge this with the else branch (line 1955..1979), which could further more reduce the number of synchronizations.

@stepinto Lines 1955–1979 also include the `pp_size == 1` storage-queue path. For `pp_size > 1`, storage queues are synchronized stage-locally because hybrid sidecar pools may differ across PP stages.

Merging that whole branch would therefore change the PP storage semantics. I’ll keep those lines unchanged and limit this PR to batching write/load completions for `pp_size > 1`.

Comment (stepinto) [reply python/sglang/srt/mem_cache/unified_radix_cache.py:2035]: Fine.  Just leave it as it.  Thank you.  cc @hzh0425

**Review by hzh0425 — APPROVED** [review]

**Comment by MichoChan** [comment]
Could you show me the startup parameters of your benchmark

**Comment by hzh0425** [comment]
@luoroger37 you should help merge the main and pass the ci

**Comment by hzh0425** [comment]
https://github.com/sgl-project/sglang/actions/runs/31989242532/job/95271967854#step:15:920

could you check the failed test?

**Comment by hzh0425** [comment]
Please keep an eye on the progress of CI yourself and ping me when （Base） has passed

**Comment by luoroger37** [comment]
Fix the CI failure caused by #31323 introducing forbidden get_server_args() reads in topk.py, which violated the global-config access ratchet.

**Comment by luoroger37** [comment]
@hzh0425  It seems all base CI has been approved.
